packages/playwright-core/src/utils/isomorphic/locatorGenerators.ts

```diff
@@ -158,19 +158,29 @@ function innerAsLocators(factory: LocatorFactory, parsed: ParsedSelector, isFram
       }
     }
     if (part.name === 'internal:control' && (part.body as string) === 'enter-frame') {
-      tokens.push([factory.generateLocator(base, 'frame', '')]);
+      const lastTokens = tokens[tokens.length - 1];
+      const lastPart = parts[index - 1];
+
+      const extendedTokens = lastTokens.map(token =>
+        factory.chainLocators([token, factory.generateLocator(base, 'frame', '')])
+      );
+      extendedTokens.push(
```

Comment:
I think we should only do this when `lastPart` is a css/xpath selector. Otherwise we'll generate something like `frameLocator('internal:attr=...')` that we don't really want to see. I wonder why don't tests expose this? Perhaps add/change some?

Reply:
The tests don't expose because they only tested `locatorOrSelectorAsSelector`, and not `asLocators`. I've added to the test to cover this.